contrib/tools/lmc.md/lgraph-search.md

```diff
@@ -130,10 +96,12 @@ fmod GRAPH-FOLDING-SEARCH is
     op isEmpty? : NodeSet -> [Bool] .
     ---------------------------------
     eq isEmpty?(.NodeSet) = true .
+```
 
-    op _[_] : Nat Fold -> NodeId [right id: .Fold prec 20] .
-    --------------------------------------------------------
+A `NodeMap` is a lookup table of `Nat |-> Node`, which supports a lookup `_[_]` and `insert` operation.
+The insert operation is aware of the user defined `Fold`s, and will not create new entries for new nodes that are less general than existing ones.
 
```

Comment:
I find the `NodeMap?` data structure name somewhat misleading. As far as I can tell, a `NodeMap?` always corresponds to a `NodeMap` --- in my mind, adding a question mark to the end sort of connotes a partial operation; AFAICT `insert` which returns one, is not partial.

I know you prefer compact names, so if a more descriptive name is out of the question, I think it would be worth it to just note that this data structure is just an intermediate used to track when fresh nodes have been added to a `NodeMap`. Also, AFACIT, the `NodeSet` argument in `NodeMap` is always either a singleton `Node` or `.NodeSet`. If that is true, I think that would be nice to note as well.

It seems that, at least here, the question mark denotes that essentially the same type but with extra metadata attached to it; `FoldedLabeledGraph?` seems to use a similar naming convention, though I don't mind as much in that case since I think that may actually not always reduce into a `FoldedLabeledGraph`, i.e. in case stepping over the frontier fails.